flutter-studio/src/io/flutter/project/ChoseProjectTypeStep.java

```diff
@@ -0,0 +1,171 @@
+import org.jetbrains.annotations.Nullable;
+
+import javax.accessibility.AccessibleContext;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static com.android.tools.idea.wizard.WizardConstants.DEFAULT_GALLERY_THUMBNAIL_SIZE;
+import static org.jetbrains.android.util.AndroidBundle.message;
+
+public class ChoseProjectTypeStep extends ModelWizardStep<FlutterProjectModel> {
+  private static final Logger LOG = Logger.getInstance(ChoseProjectTypeStep.class.getName());
+
+  private final List<ModuleGalleryEntry> myModuleGalleryEntryList;
+  private final JComponent myRootPanel;
+  private JLabel[] helpLabels;
+
+  private ASGallery<ModuleGalleryEntry> myProjectTypeGallery;
+  private Map<ModuleGalleryEntry, FlutterProjectStep> myModuleDescriptionToStepMap;
+
+  public ChoseProjectTypeStep(@NotNull FlutterProjectModel model) {
+    this(model, FlutterDescriptionProvider.getGalleryList(true));
+  }
+
+  private ChoseProjectTypeStep(@NotNull FlutterProjectModel model, @NotNull List<ModuleGalleryEntry> moduleGalleryEntries) {
+    super(model, "New Flutter Project");
+
+    myModuleGalleryEntryList = moduleGalleryEntries;
+    myRootPanel = new JPanel();
+    myRootPanel.setLayout(new VerticalFlowLayout());
+    myRootPanel.add(createGallery());
+
+    final JPanel section = new JPanel();
+    section.setLayout(new VerticalFlowLayout());
+    helpLabels = new JLabel[moduleGalleryEntries.size()];
+    int idx = 0;
+    for (ModuleGalleryEntry entry : moduleGalleryEntries) {
+      helpLabels[idx] = new JLabel(((FlutterDescriptionProvider.FlutterGalleryEntry)entry).getHelpText());
+      helpLabels[idx].setEnabled(false);
+      section.add(helpLabels[idx++]);
+    }
+
+    myProjectTypeGallery.addListSelectionListener(e -> {
+      if (e.getValueIsAdjusting()) {
+        return;
+      }
+      toggleLabel(e.getFirstIndex());
+      toggleLabel(e.getLastIndex());
+    });
+    toggleLabel(0); // Already selected.
+
+    myRootPanel.add(section);
+    FormScalingUtil.scaleComponentTree(this.getClass(), myRootPanel);
+  }
+
+  @NotNull
+  private static List<ModuleGalleryEntry> getGalleryList() {
+    return new ArrayList<>();
+  }
+
+  private void toggleLabel(int index) {
+    helpLabels[index].setEnabled(!helpLabels[index].isEnabled());
+    // TODO Delete this old code once we decide whether enabling or bolding is the way to emphasize the help text.
+    //boolean isBold = helpLabels[index].getFont().isBold();
+    //Font font = helpLabels[index].getFont().deriveFont(isBold ? Font.PLAIN : Font.BOLD);
+    //helpLabels[index].setFont(font);
+  }
+
+  @NotNull
+  @Override
+  protected JComponent getComponent() {
+    return myRootPanel;
+  }
+
+  @NotNull
+  @Override
+  public Collection<? extends ModelWizardStep> createDependentSteps() {
+    List<ModelWizardStep> allSteps = Lists.newArrayList();
```

Comment:
Old chestnut: this is not declared `final`l but above, for example, the JPanel is.  I never really drank the declare locals final doctrine but I do like consistency.

Thoughts?

(And sorry if I'm missing context --- maybe best to chat in person?)

Reply:
I'm happy to remove 'final' from the local var.